### ISSUE TITLE
[Feature] Element-wise comparison operators and boolean tensor support

### ISSUE THREAD

**Issue by brody-0125** [brody-0125/dart_tensor_preprocessing#37]
## Summary

Add element-wise comparison operations that return boolean tensors: `equal`, `notEqual`, `greater`, `greaterEqual`, `less`, `lessEqual`.

## Motivation

Comparison operators returning boolean tensors are foundational for masking, filtering, and conditional operations. They are prerequisites for `where` (conditional selection), threshold-based filtering, and attention mask generation. Every major tensor library treats these as core operations.

## Reference Implementations

- **PyTorch**: `torch.eq`, `torch.ne`, `torch.gt`, `torch.ge`, `torch.lt`, `torch.le` — [PyTorch docs](https://pytorch.org/docs/stable/generated/torch.eq.html)
- **NumPy**: `numpy.equal`, `numpy.not_equal`, `numpy.greater`, `numpy.less`, etc.
- **ONNX**: `Equal`, `Greater`, `GreaterOrEqual`, `Less`, `LessOrEqual` operators

## Verification

- ✅ `torch.eq`, `torch.gt`, etc. exist — VERIFIED
- ✅ ONNX comparison operators exist in opset 1+
- ✅ Feature is MISSING from dart_tensor_preprocessing — VERIFIED (searched for equal, greater, less, compare, boolean)
- Note: `OperationCapabilities.supportsBroadcast` flag exists but no boolean tensor operations

## Proposed API

```dart
// Element-wise comparison (returns tensor with 0.0/1.0 values)
final mask = tensor.equal(other);         // or scalar
final mask = tensor.greater(threshold);
final mask = tensor.lessEqual(other);

// As TransformOps for pipeline
final op = CompareOp.greaterThan(0.5);   // threshold comparison
final mask = op.apply(tensor);
```

## Implementation Notes

- Since `TensorBuffer` uses typed floating-point data, boolean results should be encoded as `0.0`/`1.0` in a Float32 tensor (matching ONNX Cast behavior)
- Alternatively, consider adding `DType.bool` support
- Support tensor-tensor and tensor-scalar comparisons
- SIMD-acceleratable for Float32
- Add `OperationCapabilities`:
  - `onnxOpType: 'Equal'` / `'Greater'` / `'Less'` etc.
  - `pytorchEquivalent: 'torch.eq'` / `'torch.gt'` etc.

## ROI Assessment

**High** — Foundation for masking/filtering operations. Required for `where` operation, attention masks, and threshold-based processing.
